### PR TITLE
skip CPU tests on GPU GHA jobs

### LINKED ISSUE #6957: Fix GHA Linux GPU Job not running tests on CUDA
Currently "[Unit-tests on Linux GPU](https://github.com/pytorch/vision/actions/workflows/test-linux-gpu.yml)" is not running cuda tests but supposed to.

For example, on main:
- https://github.com/pytorch/vision/actions/runs/3471795920/jobs/5801756501#step:8:17508
```
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center1-32-26-cuda] SKIPPED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-7-33-cpu] PASSED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-7-33-cuda] SKIPPED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-26-26-cpu] PASSED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-26-26-cuda] SKIPPED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-32-26-cpu] PASSED [ 53%]
test/test_functional_tensor.py::TestRotate::test_rotate[fn1-fill3-False--146-dt2-center2-32-26-cuda] SKIPPED [ 53%]
```

Context: https://github.com/pytorch/vision/pull/6804#issuecomment-1305493672

cc @seemethere

### PR DESCRIPTION
**Pull request by pmeier**
_Blocked by #6957. I've debugged why no GPU test is run until 1a2efbee6fe97a91659dc1e8ab8449e438aad4da. Please discard my commits and comments before that._

---

To save CI resources, we don't run CPU tests on GPU machines. This behavior is hardcoded to CircleCI:

https://github.com/pytorch/vision/blob/4a310f26049371959617921d0eb9b001f4d262c6/test/conftest.py#L15-L24

With the recent push to GHA (Nova), we also need this behavior there. The environment variable is called `GITHUB_ACTIONS` and is set to `"true"` in case we are in a GHA runner. This is the same as the `CIRCLECI` variable that we already handled before.

cc @seemethere

### CODE REVIEW

**Comment by pmeier** [comment]
Installation of `torchvision` on GHA [prints](https://github.com/pytorch/vision/actions/runs/3523666163/jobs/5908096488#step:8:953):

> No CUDA runtime is found, using CUDA_HOME='/work/ci_env'

**Comment by pmeier** [comment]
Looking at the collected env, it seems there is an issue with the CUDA setup.

```
Collecting environment information...
PyTorch version: 1.14.0.dev20221122
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 9.3.1 20200408 (Red Hat 9.3.1-2)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.8.15 (default, Nov  4 2022, 20:59:55)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.14.252-195.483.amzn2.x86_64-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: 11.6.124
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.3.2
/usr/local/cuda-11.6/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.3.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
```

The important bits:

```
CUDA used to build PyTorch: 11.6
[...]
Is CUDA available: False
CUDA runtime version: 11.6.124
```

It seems we install the right versions, but CUDA is not available for some reason. Maybe the driver is not set up properly?

**Comment by pmeier** [comment]
Ok, running

```
modinfo nvidia || true
nvidia-smi || true
exit 0
```

gives

```
modinfo: ERROR: Module alias nvidia not found.
/exec: line 8: nvidia-smi: command not found
```

This has nothing to do with `torchvision`. @osalpekar can you have a look?

File: test/common_utils.py [hunk 1]
```diff
@@ -13,11 +13,11 @@
 import __main__  # noqa: 401
 
 
-IN_CIRCLE_CI = os.getenv("CIRCLECI", False) == "true"
+IN_OSS_CI = any(os.getenv(var) == "true" for var in ["CIRCLECI", "GITHUB_ACTIONS"])
```
Comment (pmeier) [inline test/common_utils.py:16]: One common convention for CI providers is to set the `CI=true` environment variable. We could use that here as well, but I have no idea if that interferes with Meta internal systems. Thus, to be safe, we are explicit about the CI providers here. Given that we probably don't change them that often, I think this should be fine.

**Review by NicolasHug — APPROVED** [review]
Thank you @pmeier ! LGTM as long as it does what we want :)

**Comment by pmeier** [comment]
PR: https://github.com/pytorch/vision/actions/runs/4135402074/jobs/7147802842#step:10:35257

```
= 1 failed, 13404 passed, 17527 skipped, 6 xfailed, 31 warnings in 624.39s (0:10:24) =
```

`main`: https://github.com/pytorch/vision/actions/runs/4134781712/jobs/7146348614#step:10:35781

```
= 1 failed, 29740 passed, 1188 skipped, 9 xfailed, 246 warnings in 3832.74s (1:03:52) =
```

So roughly 6x speed-up (4x if look at the overall workflow not just the tests)

**Pull request merged**